sky/resources.py

```diff
@@ -1117,13 +1117,93 @@ def get_required_cloud_features(
+                # validation above.
+                resources_list.extend(
+                    list(Resources.from_yaml_config(new_resource_config)))
+            return resources_list
+
+        config = config.copy()
+        any_of_configs = config.pop('any_of', None)
+        ordered_configs = config.pop('ordered', None)
+        if any_of_configs is not None and ordered_configs is not None:
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError(
+                    'Cannot specify both "any_of" and "ordered" in resources.')
+
+        # Parse resources.accelerators field.
+        accelerators = config.get('accelerators')
+        if config and accelerators is not None:
+            if isinstance(accelerators, str):
+                accelerators = {accelerators}
+            elif isinstance(accelerators, dict):
+                accelerators = [
+                    f'{k}:{v}' if v is not None else f'{k}'
+                    for k, v in accelerators.items()
+                ]
+                accelerators = set(accelerators)
+            if len(accelerators) > 1 and ordered_configs:
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError(
+                        'Cannot specify multiple "accelerators" with "ordered" '
+                        'in resources.')
+            if (len(accelerators) > 1 and any_of_configs and
+                    not isinstance(accelerators, set)):
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError(
+                        'Cannot specify multiple "accelerators" with prefered '
+                        'order (i.e., list of accelerators) with "any_of" '
+                        'in resources.')
+
+        if any_of_configs:
+            resources_list = _override_resources(config, any_of_configs)
+            return set(resources_list)
+        if ordered_configs:
+            resources_list = _override_resources(config, ordered_configs)
+            return resources_list
+        # Translate accelerators field to potential multiple resources.
+        if accelerators:
+            # In yaml file, we store accelerators as a list.
+            # In Task, we store a list of resources, each with 1 accelerator.
+            # This for loop is for format conversion.
+            tmp_resources_list = []
+            for acc in accelerators:
+                tmp_resource = config.copy()
+                tmp_resource['accelerators'] = acc
+                tmp_resources_list.append(
+                    Resources._from_yaml_config_single(tmp_resource))
+
+            if isinstance(accelerators, (list, set)):
+                return type(accelerators)(tmp_resources_list)
+            else:
+                with ux_utils.print_exception_no_traceback():
+                    raise RuntimeError('Accelerators must be a list or a set.')
```

Comment:
```suggestion
                    raise RuntimeError('Accelerators must be a list or a set when multiple accelerators are specified.')
```

Reply:
Good call! Actually, I find this should be assertion instead of raise, as the type of the resources should have ben guaranteed by the schema check.